### PR TITLE
[GP5-10] Product - As a user, I want to see a detailed information of a product

### PR DESCRIPTION
**Pull request by BarsMoxxi**

### CODE REVIEW

**Comment by haikal1223** [comment]
ini belom sampe masukin barang ke cart ya?

File: projects/client/src/pages/Product/index.js [hunk 1]
```diff
@@ -1,132 +1,79 @@
...
-      {stock > 0 ? (
-        <Button onClick={handleAddOrder}> This is product page {id}</Button>
-      ) : (
-        <Button isDisabled="true"> This is product page {id}</Button>
-      )}
-      <Toaster />
-    </>
-  );
+    <div>
+      <Card maxW='sm'>
```
Comment (rohmatsetiawanmta) [inline projects/client/src/pages/Product/index.js:42]: **line 42**
maxW='sm' jadi w='full' aja supaya tampilan cartnya full

File: projects/client/src/pages/Product/index.js [hunk 2]
```diff
@@ -1,132 +1,79 @@
...
+        <Box borderWidth='1px' borderRadius='lg' overflow='hidden'>
+
+          <Image h='200px' w='full' p='2'
+            objectFit='contain' src={productData.image_url} />
+
+          {productData.name}
```
Comment (rohmatsetiawanmta) [inline projects/client/src/pages/Product/index.js:48]: **line 48**
better dibuat align center, jadi gini aja

<Text align='center'>
    {productData.name}
</Text>

**Review by rohmatsetiawanmta — COMMENTED** [review]
check Product index.js line 42 and 48

**Comment by haikal1223** [comment]
resolve conflict

File: projects/client/src/pages/Product/index.js [hunk 3]
```diff
@@ -1,135 +1,84 @@
...
+          <Image h='200px' w='full' p='2'
+            objectFit='contain' src={productData.image_url} />
+
+          <Text align='center'>
+            {productData.name}
+          </Text>
+          <Text align='center'>
+            {productData.price?.toLocaleString('id-ID', { style: 'currency', currency: 'IDR' })}
+          </Text>
+
```
Comment (doomsdoge) [inline projects/client/src/pages/Product/index.js:54]: Bisa add stock & category mungkin

File: projects/client/src/pages/Product/index.js [hunk 4]
```diff
@@ -1,135 +1,84 @@
...
+                  variant='outline'
+                  isDisabled={quantity === 1}
+                />
+                <Button w='full' bg={"#5D5FEF"} color='white' _hover={{ color: 'blue.500', bg: "white", border: '1px solid skyblue' }} >ADD TO CART ({quantity})</Button>
+                <IconButton
+                  aria-label='increase quantity'
+                  icon={<AddIcon />}
+                  onClick={() => handleQuantityChange('increase')}
+                  size='sm'
+                  variant='outline'
```
Comment (doomsdoge) [inline projects/client/src/pages/Product/index.js:72]: add isDisabled as in decrement button. But for this one, set to disabled when quantity >= stock available

**Review by RaihanAdity — APPROVED** [review]

File: projects/client/src/pages/Product/index.js (hunk 1 above)
Comment (BarsMoxxi) [reply projects/client/src/pages/Product/index.js]: done

File: projects/client/src/pages/Product/index.js (hunk 2 above)
Comment (BarsMoxxi) [reply projects/client/src/pages/Product/index.js]: done

File: projects/client/src/pages/Product/index.js [hunk 5]
```diff
@@ -1,132 +1,79 @@
...
-  );
+    <div>
+      <Card maxW='sm'>
+        <Box borderWidth='1px' borderRadius='lg' overflow='hidden'>
+
+          <Image h='200px' w='full' p='2'
+            objectFit='contain' src={productData.image_url} />
+
+          {productData.name}
+
```
[the remark this answers is not in the document]
Comment (BarsMoxxi) [reply projects/client/src/pages/Product/index.js:54]: done

File: projects/server/controller/productController.js [hunk 6]
```diff
@@ -92,35 +92,35 @@ module.exports = {
...
-        where: {
-          id,
-        },
-      });
-      if (findProduct)
-        return res.status(200).send({
-          isError: false,
-          message: "Data is found",
-          data: findProduct,
+        const findProduct = await db.product.findAll({
```
[the remark this answers is not in the document]
Comment (BarsMoxxi) [reply projects/server/controller/productController.js:98]: done

File: projects/server/controller/productController.js [hunk 7]
```diff
@@ -92,35 +92,35 @@ module.exports = {
...
+            where: {
+                id,
+            },
+        });
```
[the remark this answers is not in the document]
Comment (BarsMoxxi) [reply projects/server/controller/productController.js:102]: done

**Review by haikal1223 — APPROVED** [review]

**Review by doomsdoge — APPROVED** [review]

**Pull request merged**